diff_cover/diff_quality_tool.py

```diff
@@ -222,20 +224,33 @@ def main(argv=None, directory=None):
         last_char = user_options[-1]
         if first_char == last_char and first_char in ('"', "'"):
             user_options = user_options[1:-1]
+    reporter = None
     driver = QUALITY_DRIVERS.get(tool)
-
-    if driver is not None:
-        # If we've been given pre-generated reports,
-        # try to open the files
+    if driver is None:
+        # The requested tool is not built into diff_cover. See if another Python
+        # package provides it.
+        pm = pluggy.PluginManager('diff_cover')
+        pm.add_hookspecs(hookspecs)
+        pm.load_setuptools_entrypoints('diff_cover')
+        for name, reporter_fn in pm.hook.diff_cover_report_quality():
+            if name == tool:
+                reporter = reporter_fn()
+                break
+
+    if reporter is not None or driver is not None:
```

Comment:
Could this just be `if reporter or driver`?

Reply:
That's true. I began my career in C and it left me overly fussy about Boolean expressions. 🙂 I'll change it.